File: api-tester/src/main/java/com/revenuecat/apitester/java/GoogleReplacementModeAPI.java
package com.revenuecat.apitester.java;

import com.revenuecat.purchases.models.GoogleReplacementMode;

@SuppressWarnings({"unused"})
final class GoogleReplacementModeAPI {
    static void check(final GoogleReplacementMode mode) {
        switch (mode) {
            case WITHOUT_PRORATION:
            case WITH_TIME_PRORATION:
            case CHARGE_FULL_PRICE:
            case CHARGE_PRORATED_PRICE:
            case DEFERRED:
        }
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/GoogleReplacementModeAPI.kt
package com.revenuecat.apitester.kotlin

import com.revenuecat.purchases.models.GoogleReplacementMode

@Suppress("unused", "UNUSED_VARIABLE")
private class GoogleReplacementModeAPI {
    fun check(mode: GoogleReplacementMode) {
        when (mode) {
            GoogleReplacementMode.WITHOUT_PRORATION,
            GoogleReplacementMode.WITH_TIME_PRORATION,
            // GoogleReplacementMode.DEFERRED,
            GoogleReplacementMode.CHARGE_FULL_PRICE,
            GoogleReplacementMode.CHARGE_PRORATED_PRICE,
            -> {}
        }.exhaustive
    }
}
